### ISSUE TITLE
edd_is_cart_saved logic issue

### ISSUE THREAD

**Issue by TimBHowe** [awesomemotive/easy-digital-downloads#9658]
### Expected behavior
When I call the `edd_is_cart_saved` function to see if a saved the cart for a logged in user matches that same users session cart without making any changes to the cart it should return `bool( true )`.

https://github.com/awesomemotive/easy-digital-downloads/blob/ac99f24ffd5dcdc41992eeecb1720af69e50e13e/includes/cart/functions.php#L581

### Actual behavior
Looking at the Cart class function it called it seems the logic is reversed here 
https://github.com/awesomemotive/easy-digital-downloads/blob/ac99f24ffd5dcdc41992eeecb1720af69e50e13e/includes/cart/class-edd-cart.php#L1352

Where it checks if the  `$saved_cart`  from the `$saved_cart = get_user_meta( get_current_user_id(), 'edd_saved_cart', true );`
and compares it against the  `EDD()->session->get( 'edd_cart' ) `  and if the are equal it is false when it should probably be when they are **_not_** equal it is false

### Steps to reproduce the behavior
1) Enable cart saving on checkout
2) Create a cart as a logged in user and save it
3) var_dump the reply from `edd_is_cart_saved();`

### Information (if a specific version is affected):
PHP Version: 8.1

EDD Version (or branch): 3.1.2

WordPress Version: 6.2.2

Any other relevant information:
Here is some testing code
```php
function edd_testing_is_cart_saved() {
	var_dump( edd_is_cart_saved() );
}
add_action('wp_footer','edd_testing_is_cart_saved');
```
```php
	/**
	 * Checks if the cart has been saved
	 *
	 * @since 2.7
	 * @return bool
	 */
	public function is_saved() {
		if ( ! $this->is_saving_enabled() ) {
			var_dump(1);
			return false;
		}

		$saved_cart = get_user_meta( get_current_user_id(), 'edd_saved_cart', true );

		if ( is_user_logged_in() ) {
			if ( ! $saved_cart ) {
				var_dump(2);
				return false;
			}

			if ( $saved_cart === EDD()->session->get( 'edd_cart' ) ) {
				var_dump(3);
				return false;
			}
			var_dump(4);
			return true;
		} else {
			if ( ! isset( $_COOKIE['edd_saved_cart'] ) ) {
				var_dump(5);
				return false;
			}

			if ( json_decode( stripslashes( $_COOKIE['edd_saved_cart'] ), true ) === EDD()->session->get( 'edd_cart' ) ) {
				var_dump(6);
				return false;
			}

			var_dump(7);
			return true;
		}
	}
	```

**Comment by TimBHowe** [comment]
https://github.com/awesomemotive/easy-digital-downloads/pull/9659

**Comment by cklosowski** [comment]
Thanks for the report @TimBHowe We'll take a look at this.

@robincornett can you triage this and get it into our board for release triage.

**Comment by cklosowski** [comment]
Thanks @TimBHowe This was fixed in EDD 3.2.0.
